docs/UserGuide.md

```diff
@@ -16,21 +23,85 @@ Cobb can also run on any machine, *without the need for an internet connection*.
+   (external link to Oracle) provides a step-by-step installation guide for Java, if needed.
 
-2. Download the latest `cobb.jar` from [here](https://github.com/AY2223S1-CS2103T-F12-1/tp/releases).
+2. Download the latest `cobb.jar` from our [release page](https://github.com/AY2223S1-CS2103T-F12-1/tp/releases).
 
-3. Copy the file to the folder you want to use as the _home folder_ for Cobb.
+3. Copy the file to the folder you want to use as the _home folder_ for Cobb. All data will be created and stored
+   within this directory.
 
-4. Double-click the file to start the app. The GUI similar to the below should appear in a few seconds. Note how the app contains some sample data.<br>
+4. Double-click the file to start the app. A window similar to the below should appear in a few seconds. Note how the application contains some sample data.<br>
    ![Ui](images/Ui.png)
 
-5. Type the command in the command box and press Enter to execute it. e.g. typing **`help`** and pressing Enter will display some help text.<br>
-   Some example commands you can try:
+Congratulations! Cobb is now set up and ready to work on your system.
+
+If you encounter any bugs during the setup process, do check out the [FAQ](#faq) section of this guide, which hopefully
+contains some information that can help you diagnose your issue.
+
+<div markdown="span" class="alert alert-warning">:exclamation: **Caution:**
+On first launch, Cobb will create a few files that have the extension `.json` in its *home directory*. These files are used
+by Cobb to store its data. **Edit these at your own risk**, as Cobb will start with an empty database if it detects any
+error in the formatting of the data in these files.
+</div>
+
+## Key definitions
+
+This section aims to define some terms that pop up throughout the user guide. These terms are used often, so it would be 
+good to take note of them before continuing.
+
+### Command-specific terms
+1. **Command**: A piece of instruction that a user types into the command input box to instruct Cobb's behaviour.
+   Commands have many variations, and can serve many purposes.
+2. **Syntax**: The manner in which is a command is to be typed. Can be thought of as the "format" of the command.
+3. **Parameter**: An input to a command. Can be thought of as a piece of information a user provides that determines the
+   behaviour of a specific command.
+4. **Command string**: The text that forms the command.
+5. **Flag**: A marker that signifies the start of a parameter to a command. For Cobb, in the command `findbuyer -k Tim`, 
+   `-k` is the flag, and `Tim` is the parameter.
+6. **Fuzzy**: A flag that signifies loose filtering, that is, given a set of filter requirements, any user that matches at least one
+   the given requirements will be returned.
+7. **Strict**: A flag that signifies tight filtering, that is, given a set of filter requirements, any user matching all of the
+   given requirements will be returned.
```

Comment:
I think this one can just explain in the filter and match command, since its only used there